File: University Projects/C# Projects/Design and Programming Environments/Proiect/persistence/OrganizersDBRepository.cs
using model;
using Serilog;
using System;
using System.Collections.Generic;
using System.Text;

namespace persistence
{
    public class OrganizersDBRepository : IOrganizerRepository
    {
        private readonly DBUtils _dbUtils;
        private readonly ILogger _logger = Log.ForContext<OrganizersDBRepository>();

        public OrganizersDBRepository(DBUtils dbUtils)
        {
            _logger.Information("Initializing OrganizersDBRepository");
            _dbUtils = dbUtils;
        }

        public Organizer FindByUsernameAndPassword(string username, string password)
        {
            _logger.Information("Validating username and password for user: {Username}", username);

            var con = _dbUtils.GetConnection();
            string query = "SELECT id FROM Organizers WHERE username = @User AND password = @Pass";

            try
            {
                using var stmt = con.CreateCommand();
                stmt.CommandText = query;

                var pUser = stmt.CreateParameter();
                pUser.ParameterName = "@User";
                pUser.Value = username;
                stmt.Parameters.Add(pUser);

                var pPass = stmt.CreateParameter();
                pPass.ParameterName = "@Pass";
                pPass.Value = password;
                stmt.Parameters.Add(pPass);

                using var result = stmt.ExecuteReader();
                if (result.Read())
                {
                    int id = result.GetInt32(0);
                    _logger.Information("Credentials valid for user: {Username} with ID: {Id}", username, id);
                    return new Organizer(id, username, password);
                } else
                {
                    _logger.Warning("Invalid credentials for user: {Username}", username);
                    return null;
                }
            }
            catch (Exception e)
            {
                _logger.Error(e, "Database error validating credentials");
                Console.WriteLine("Error DB " + e.Message);
                return null;
            }
        }

        public void Save(Organizer entity) => throw new NotImplementedException();
        public void Delete(int id) => throw new NotImplementedException();
        public void Update(Organizer entity) => throw new NotImplementedException();
        public Organizer FindById(int id) => throw new NotImplementedException();
        public List<Organizer> FindAll() => throw new NotImplementedException();
    }
}

File: University Projects/Java Projects/Design and Programming Environments/Proiect/Persistence/src/main/java/app/persistance/repository/OrganizersDBRepository.java
package app.persistance.repository;

import app.persistance.OrganizerRepository;

import app.model.Organizer;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.util.List;

public class OrganizersDBRepository implements OrganizerRepository {
    private final DBUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public OrganizersDBRepository(DBUtils dbUtils) {
        logger.info("Initializing OrganizersDBRepository with utils: {} ", dbUtils);
        this.dbUtils = dbUtils;
    }

    @Override
    public Organizer findByUsernameAndPassword(String username, String password) {
        logger.info("Validating username and password: {} {}", username, password);

        var con = dbUtils.getConnection();
        String query = "SELECT id FROM Organizers WHERE username = ? AND password = ?";
        try (var stmt = con.prepareStatement(query)) {
            stmt.setString(1, username);
            stmt.setString(2, password);
            try (var result = stmt.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    logger.info("Found organizer with ID: {}", id);
                    return new Organizer(id, username, password);
                } else {
                    logger.info("No organizer found with provided credentials");
                    return null;
                }
            }
        } catch (Exception e) {
            logger.error(e);
            System.out.println("Error DB " + e);
            return null;
        }
    }

    @Override
    public void save(Organizer entity) {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public void delete(Integer integer) {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public void update(Organizer entity) {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public Organizer findById(Integer integer) {
        throw new UnsupportedOperationException("Not implemented yet");
    }

    @Override
    public List<Organizer> findAll() {
        throw new UnsupportedOperationException("Not implemented yet");
    }
}
